### PR TITLE
Set context name in status report based on app env

### PR DESCRIPTION
**Pull request by jseppi**
This PR adds a new app config value called `app_env` that is derived from the `APP_ENV` environment variable. It defaults to `'development'`. This setting is used to set the GitHub status notification "context" to a name based on that setting, such as `federalist-staging/build`. For production, we still use the nicer `federalist/built` value.

To Do:

- [x] add test for status context based on app env

Ref #956

### CODE REVIEW

File: api/services/GithubBuildStatusReporter.js [hunk 1]
```diff
@@ -1,104 +1,106 @@
...
-    site = model
-    return loadBuildUserAccessToken(build)
-  }).then(accessToken => {
-    githubClient = authenticateGithubClient(accessToken)
+    site = model;
+    return loadBuildUserAccessToken(build);
+  }).then((accessToken) => {
+    const githubClient = authenticateGithubClient(accessToken);
+
+    const context = environment.env === 'production'
```
Comment (jseppi) [inline api/services/GithubBuildStatusReporter.js]: There's a lot of noise in the diff due to linting, but this is the relevant change.

Comment (jmhooper) [inline api/services/GithubBuildStatusReporter.js]: I am not sure that the `environment` service exposes that `env` var. That is used primarily to get values from the `federalist-${environment}-env` user provided service.

Instead, we may want to consider adding either an `app_env` config to `config/app.js`, or a `github_status_context` config to `config/build.js`.

**Comment by jseppi** [comment]
I went with checking `process.env.APP_ENV` in GithubBuildStatusReporter.js. I had a hard time working with the tests to instead make it part of `config.app`, but I think that would be a better place to set it (and provide a default value).

**Comment by jseppi** [comment]
I don't know why I was thinking I couldn't do the same thing I was doing with `process.env.APP_ENV` for `config.app.app_env`. Disregard previous comment :)

**Pull request merged**